### PR TITLE
MONGOCRYPT-606 publish Debian arm64 packages

### PR DESCRIPTION
**Pull request by kevinAlbs**
# Summary

Publish Debian 11, 10, and 9.2 packages for arm64.

## Other improvements

The URLs for the public key are updated in the install instructions. `www.mongodb.org/static/pgp` is replaced with `pgp.mongodb.com`. The old URL redirected to the new URL. Other [public documentation](https://www.mongodb.com/docs/manual/tutorial/verify-mongodb-packages/#download-then-import-the-key-file) also refers to `pgp.mongodb.com`.

The `download distro package tarball` Evergreen function is removed. The package tarball is referenced by URL in the `curator` command. I expect there is no need to also download the tarball within the Evergreen task.

## Testing

A [patch build](https://spruce.mongodb.com/version/65bbde24c9ec44bdd8bcd093) was run with the `"${is_patch}" = "true"` skips in `publish packages` and `setup packaging credentials` temporarily commented out to test publishing the new development packages.

This PR adds an Earthly target `test-deb-packages-from-ppa` to test installing of the `.deb` packages.

The published packages were tested locally on arm64 macOS to match the architecture:
```bash
./.evergreen/earthly.sh +test-deb-packages-from-ppa --env=deb11 --distro="debian bullseye"
./.evergreen/earthly.sh +test-deb-packages-from-ppa --env=deb10 --distro="debian buster"
./.evergreen/earthly.sh +test-deb-packages-from-ppa --env=deb9 --distro="debian stretch"
```

### CODE REVIEW

**Review by rcsanchez97 — APPROVED** [review]
LGTM

File: Earthfile [hunk 1]
```diff
@@ -372,3 +377,58 @@ build:
...
+#     · "development" packages are updated by the `publish-packages` tasks every commit.
+test-deb-packages-from-ppa:
+    ARG env="deb11"
+    ARG distro="debian bullseye"
+    ARG version="development"
+    FROM +env.$env
+    WORKDIR /s
+    RUN __install apt-transport-https
+    # Install libmongocrypt following install steps described in README.md:
+    RUN sh -c 'curl -s --location https://pgp.mongodb.com/libmongocrypt.asc | gpg --dearmor >/etc/apt/trusted.gpg.d/libmongocrypt.gpg'
```
Comment (vector-of-bool) [inline Earthfile:420]:
```suggestion
    RUN curl -s --location https://pgp.mongodb.com/libmongocrypt.asc | gpg --dearmor >/etc/apt/trusted.gpg.d/libmongocrypt.gpg
```

Wrapping in `sh -c` is not necessary, as all `RUN` lines are implicitly executed by `sh -c`

File: Earthfile [hunk 2]
```diff
@@ -372,3 +377,58 @@ build:
...
+    RUN echo "deb https://libmongocrypt.s3.amazonaws.com/apt/$distro/libmongocrypt/$version main" | tee /etc/apt/sources.list.d/libmongocrypt.list
+    RUN apt-get update
+    RUN apt-get install -y libmongocrypt-dev
+    # Test using libmongocrypt:
+    RUN __install gcc
```
Comment (vector-of-bool) [inline Earthfile:425]:
```suggestion
    # Test using libmongocrypt:
    RUN __install libmongocrypt-dev gcc
```

**Review by vector-of-bool — APPROVED** [review]
Apologies for the lengthy delay. Only two minor tweaks are recommended, but LGTM.

**Pull request merged**